serialization/serialization_config.go

```diff
@@ -62,3 +74,30 @@ func (c Config) Clone() Config {
 		ClassDefinitions:                    defs,
 	}
 }
+
+func (c Config) Validate() error {
+	return nil
```

Comment:
Since the Config is public, it can be configured element by element right? So, in that case should we validate things like `PortableFactories` (i.e. by checking the factory ids)

Reply:
Right. I've refactored `PortableFactories` so it's a slice instead of map, so factory ID is can only be specified in the factory: https://github.com/hazelcast/hazelcast-go-client/pull/561/commits/c18e9a797f3fe4e3757bafe0955fc76a374783b8